### PR TITLE
[WFLY-16917] Upgrade to Jandex 3.0.0

### PR DESCRIPTION
**Pull request by Ladicek**
https://issues.redhat.com/browse/WFLY-16917

Marking as a draft as this is not ready yet.

### CODE REVIEW

**Comment by yersan** [comment]
@Ladicek We need to have this PR without conflicts in other to merge the wildfly-core counterpart. Could you take a look at the conflicts and confirm when you are ready for this one? thanks!

**Comment by yersan** [comment]
Requires https://github.com/wildfly/wildfly-core/pull/5190

**Comment by Ladicek** [comment]
Ah, let me rebase then.

**Comment by Ladicek** [comment]
Rebased.

**Comment by yersan** [comment]
Hi @Ladicek , besides the required WildFly Core upgrade, is now this PR ready to test?

**Comment by Ladicek** [comment]
@yersan Yes.

File: pom.xml [hunk 1]
```diff
@@ -577,7 +577,7 @@
         <version.org.testcontainers>1.16.0</version.org.testcontainers>
         <version.org.testng>7.4.0</version.org.testng>
         <version.org.wildfly.arquillian>5.0.0.Alpha5</version.org.wildfly.arquillian>
-        <version.org.wildfly.core>19.0.0.Beta17</version.org.wildfly.core>
+        <version.org.wildfly.core>19.0.0.Beta18-SNAPSHOT</version.org.wildfly.core>
```
Comment (yersan) [inline pom.xml:580]: Would you mind removing the wildfly-core upgrade from this PR? It is not part of it

**Review by yersan — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (Ladicek) [reply pom.xml]: Done.

**Review by yersan — APPROVED** [review]
LGTM. @Ladicek I think you can convert it now to a regular PR instead of a draft.

**Comment by Ladicek** [comment]
Marked as ready.

**Review by bstansberry — APPROVED** [review]

**Review by pferraro — APPROVED** [review]

**Comment by yersan** [comment]
Superseded by: https://github.com/wildfly/wildfly/pull/16083

**Comment by bstansberry** [comment]
Merged as part of #16083.

Thanks, @Ladicek!

**Pull request closed without merging**